applications/blackhole/src/blackhole_socket_callback.erl

```diff
@@ -55,6 +59,24 @@ recv(_SessionPid, SessionId, Message, Context) ->
     lager:info("receive unknown message ~p on socket ~p", [Message, SessionId]),
     {'ok', Context}.
 
+amqp_send(Context, Data) ->
+    case blackhole_util:is_authorized(Context) of
+        'true' ->
+            {Message} = Data,
+            {Keys, _} = lists:unzip(Message),
+
+            SendMessage = Message ++ [
+                {<<"Msg-ID">>, wh_util:rand_hex_binary(16)}
+                | wh_api:default_headers(<<"qubicle">>, <<"1.0">>)
+            ],
+
+            {'ok', Payload} = wh_api:build_message(SendMessage, [], Keys),
+            amqp_util:basic_publish(<<"qubicle">>, <<"qubicle.recipient">>, Payload, ?DEFAULT_CONTENT_TYPE);
```

Comment:
This seems awfully specific to qubicle; maybe is should be part of `bh_qubicle`?